upup/pkg/fi/nodeup/nodetasks/service.go

```diff
@@ -67,11 +72,22 @@ func (p *Service) GetDependencies(tasks map[string]fi.Task) []fi.Task {
 		// LoadImageTask or IssueCert. If there are any LoadImageTasks (e.g. we're
 		// launching a custom Kubernetes build), they all depend on
 		// the "docker.service" Service task.
-		switch v.(type) {
-		case *File, *Package, *UpdatePackages, *UserTask, *GroupTask, *Chattr, *BindMount, *Archive:
+		switch v := v.(type) {
+		case *Package, *UpdatePackages, *UserTask, *GroupTask, *Chattr, *BindMount, *Archive:
 			deps = append(deps, v)
-		case *Service, *LoadImageTask, *IssueCert:
+		case *Service, *LoadImageTask, *IssueCert, *BootstrapClient:
 			// ignore
+		case *File:
+			// Skip adding kubeconfig files as dependencies for services other than kubelet
+			if strings.HasSuffix(v.Path, "/kubeconfig") && p.Name != kubeletService {
+				continue
+			}
+			deps = append(deps, v)
+		case *KubeConfig:
+			// Only add kubeconfigs as dependencies for kubelet
```

Comment:
It's not really necessary to add this as a dependency, since the `File` that `kubeletService` depends on will pull this in as a transitive dependency.

Reply:
Agreed, moved to the ignore list.